packages/udt/README.md

```diff
@@ -36,66 +36,143 @@
+
+```ts
+const pudt = new ccc.udt.UdtPausable(
+  udtScriptCell.outPoint,
+  udtType,
   {
-    codeHash: "0x...",
-    hashType: "type",
-    args: "0x...",
+    executor,
   },
 );
+const pudtEnumeratePaused = await pudt.enumeratePaused();
 ```
 
-You can directly call the methods in the script:
+## Advanced Usage
+
+1. Some of the methods allows you to generate a transaction object directly while taking care of most of the details for you. You just need to follow the guidance of the docs provided via your IDE.
 
 ```ts
-const { res: udtSymbol } = await udt.symbol();
-const { res: pauseList } = await udtPausable.enumeratePaused();
+const receiverA =
+  "ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsq2jk6pyw9vlnfakx7vp4t5lxg0lzvvsp3c5adflu";
+
+const { script: lockA } = await ccc.Address.fromString(
+  receiverA,
+  signer.client
+);
+
+const udtTransferTx = (
+  await udt.transfer(signer, [
+    {
+      to: lockA,
+      amount: 10000,
+    },
+  ])
+).res;
 ```
 
-Some of the methods can return a `ccc.Transaction`. For example, you can call `transfer` with the following params:
+Many of these methods also allow you to pass in a previous `ccc.TransactionLike` object as the second argument, which allows you for example to transfer multiple UDT cells in a single transaction.
 
 ```ts
-const { script: to } = await signer.getRecommendedAddressObj();
+const receiverB =
+  "ckt1qzda0cr08m85hc8jlnfp3zer7xulejywt49kt2rr0vthywaa50xwsqflz4emgssc6nqj4yv3nfv2sca7g9dzhscgmg28x";
+const { script: lockB } = await ccc.Address.fromString(
+  receiverB,
+  signer.client
+);
+let combinedTransferTx = (
+  await udt.transfer(
+    signer,
+    [
+      {
+        to: lockB,
+        amount: 20000,
+      },
+    ],
+    udtTransferTx
+  )
+).res;
+```
+
+2. You only need to complete the inputs of the transaction just like processing any other transactions with CCC.
 
-const { res: transferTx } = await udt.transfer(signer, [{ to, amount: 100 }]);
-const completedTx = await udt.completeUdtBy(transferTx, signer);
+```ts
+// Note: You need to connect your wallet for the following parts. You also need to have enough balance of the specified UDT in your wallet.
+combinedTransferTx = await udt.completeBy(combinedTransferTx, signer);
+await combinedTransferTx.completeFeeBy(signer);
+await render(combinedTransferTx);
+const combinedTransferTxHash = await signer.sendTransaction(combinedTransferTx);
 
-await completedTx.completeInputsByCapacity(signer);
-await completedTx.completeFeeBy(signer);
-const transferTxHash = await signer.sendTransaction(completedTx);
+console.log(combinedTransferTxHash);
 ```
 
+Full runnable example can be found at [here](https://live.ckbccc.com/?src=nostr:nevent1qqs8q20jvxqfsrhqw4te248qduex39dgls7qajhuc42kale0yqatdhspzemhxue69uhhyetvv9ujumn0wd68ytnzv9hxgqg5waehxw309ahx7um5wghx77r5wghxgetkqy28wumn8ghj7un9d3shjtnyv9kh2uewd9hspusjlf)
```

Comment:
Nostr event links should not be used in documents since they can expire at any time. Use permanent links instead, e.g. from GitHub raw files.